src/internet_identity/src/main.rs

```diff
@@ -813,6 +813,31 @@ fn security_headers() -> Vec<HeaderField> {
     vec![
         ("X-Frame-Options".to_string(), "DENY".to_string()),
         ("X-Content-Type-Options".to_string(), "nosniff".to_string()),
+
+        // Note: trusted types cannot be used due to firefox not supporting SRI.
+        // See https://bugzilla.mozilla.org/show_bug.cgi?id=1409200.
+        //
+        // script-src 'unsafe-eval' is required because agent-js uses a WebAssembly module for bls signatures.
+        // There is currently no other way to allow execution of WebAssembly modules with CSP.
+        // See https://github.com/WebAssembly/content-security-policy/blob/main/proposals/CSP.md and
+        // https://dfinity.atlassian.net/browse/FOLLOW-383.
+        //
+        // script-src 'unsafe-inline' https: are only there for backwards compatibility and ignored
+        // by modern browsers.
+        (
```

Comment:
Can you clarify why we need those headers, and what they do? Where did you get this list from? Is this also https://owasp.org/www-project-secure-headers/ ?

Reply:
A good overview over CSP directives is provided here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
As additional resources, I used https://csp.withgoogle.com/docs/index.html and https://csp-evaluator.withgoogle.com/.
In general, they restrict which resources can be used within the execution context of Internet Identity.